src/main/java/com/kumuluz/ee/rest/client/mp/spec/RestClientBuilderImpl.java

```diff
@@ -317,6 +320,23 @@ public RestClientBuilder hostnameVerifier(HostnameVerifier hostnameVerifier) {
                     .remove(WWWAuthenticationProtocolHandler.NAME);
         }
 
+        if (ConfigurationUtil.getInstance().get("kumuluzee.rest-client.proxy.url").isPresent()){
+            String proxyURI = ConfigurationUtil.getInstance().get("kumuluzee.rest-client.proxy.url").get();
+            try{
+                URI u = new URI(proxyURI);
+                JettyConnectorProvider.getHttpClient(client).getProxyConfiguration().getProxies().add(new HttpProxy(u.getHost(),u.getPort()));
```

Comment:
Can you try setting these properties using Jersey configuration instead of Jetty?

For proxy URL see: https://github.com/jersey/jersey/blob/faa809da43538ce31076b50f969b4bd64caa5ac9/connectors/jetty-connector/src/main/java/org/glassfish/jersey/jetty/connector/JettyConnector.java#L195

For username/password see: https://github.com/jersey/jersey/blob/faa809da43538ce31076b50f969b4bd64caa5ac9/connectors/jetty-connector/src/main/java/org/glassfish/jersey/jetty/connector/JettyConnector.java#L190

Add them using `clientBuilder.property(...)`.

Reply:
Currently the JettyConnector does not support proxy authorization.
https://github.com/eclipse-ee4j/jersey/issues/3996
To add this through Jersey configuration we'd have to change to another connect (e.g ApacheConnector).
I'm working to add this support to JettyConnector, but it should take a while to be available on a release.

I'm going to change the proxy URL to use the Jersey configuration. How do you think we should proceed to the auth data?

Reply:
Doesn't `JettyClientProperties.PREEMPTIVE_BASIC_AUTHENTICATION` (second link) do the same thing that you are doing now through `JettyConnectorProvider.getHttpClient(client).getAuthenticationStore() ...`?

If that doesn't work just leave it as is and we will fix it when JettyConnector gets updated.

Reply:
Using `JettyClientProperties.PREEMPTIVE_BASIC_AUTHENTICATION` as bellow should works but it resolves to null on the JettyConnector. Any tips why?
                    `clientBuilder.property(JettyClientProperties.PREEMPTIVE_BASIC_AUTHENTICATION,new BasicAuthentication(new URI(proxyURI), "<<ANY_REALM>>", proxyUser, proxyPass));`

Anyway, the Jersey has two configs to proxy authorization, maybe using `PREEMPTIVE_BASIC_AUTHENTICATION` is not best way.

I commited the change to use the Jersey for the proxy URL.